activesupport/lib/active_support/xml_mini.rb

```diff
@@ -62,22 +62,17 @@ def content_type
       "yaml"     => Proc.new { |yaml| yaml.to_yaml }
     } unless defined?(FORMATTING)
 
-    # TODO use regexp instead of Date.parse
+    DATE_REGEX = /\A(\d{4})-(\d{2})-(\d{2})\z/ unless defined?(DATE_REGEX)
+
     unless defined?(PARSING)
       PARSING = {
         "symbol"       => Proc.new { |symbol|  symbol.to_s.to_sym },
-        "date"         => Proc.new { |date|    ::Date.parse(date) },
+        "date"         => Proc.new { |date|    (m = DATE_REGEX.match(date)) ? ::Date.new(m[1].to_i, m[2].to_i, m[3].to_i) : ::Date.parse(date) },
         "datetime"     => Proc.new { |time|    Time.xmlschema(time).utc rescue ::DateTime.parse(time).utc },
         "duration"     => Proc.new { |duration| Duration.parse(duration) },
         "integer"      => Proc.new { |integer| integer.to_i },
         "float"        => Proc.new { |float|   float.to_f },
-        "decimal"      => Proc.new do |number|
```

Comment:
As mentioned, this looks like a cosmetic change which should not be included in a bug fix. My eyes could be deceiving me though :joy:

Reply:
> if I submitted a different pull request with just this cosmetic change, would it be accepted?

No, but it's up to a core member if they're willing to accept that trade-off, for example if it's apart of a bigger refactor.

Just going by what the existing policy is, documented here:
https://edgeguides.rubyonrails.org/contributing_to_ruby_on_rails.html#write-your-code

> Changes that are cosmetic and do not add anything substantial to the stability, functionality, or testability of Rails will generally not be accepted (read more about [our rationale behind this decision](https://github.com/rails/rails/pull/13771#issuecomment-32746700)).